### DISCUSSION TITLE
Error compiling MOOSE and MASTODON on Falcon

### DISCUSSION THREAD

**Discussion by samyogshr** [idaholab/moose#16793] [Q&A General] [answered]
Hi Moose group,

I am using these commands to build MASTODON and MOOSE on Falcon but I get error while compiling. How can I resolve this? I have attached outcome of module list, if it helps.

**Scripts used:**
mkdir ~/projects4
cd ~/projects4
git clone https://github.com/idaholab/mastodon.git
cd mastodon
git checkout master
git submodule update --init moose

cd moose/scripts/
JOBS=8 ./update_and_rebuild_libmesh.sh
cd ~/projects4/mastodon

make -j 8
./run_tests -j 8

**Error I get when compiling:**
[/home/shresamy/projects5/mastodon/moose/framework/build/header_symlinks/Moose.h:236:2: error: #error PETSc has not been detected, please ensure your environment is set up properly then rerun the libmesh build script and try to compile MOOSE again.
 #error PETSc has not been detected, please ensure your environment is set up properly then rerun the libmesh build script and try to compile MOOSE again.

fatal error: petscsys.h: No such file or directory

**Module list:**
![modulelist](https://user-images.githubusercontent.com/48602980/105542291-17a70080-5cc7-11eb-9ae8-c9b20ae553e3.PNG)

Thank you,
Samyog

**Comment by aeslaughter** [comment, marked as answer]
@milljm You have any thoughts on this?

**Reply by samyogshr** [reply]
@aeslaughter Thank you but this has been resolved,

**Reply by GiudGiud** [reply]
Marking it as solved so it doesn't show up when looking for unsolved questions.
Please consider posting what the solution to your problem was (and tagging it as the accepted answer) to help others in the future.

**Reply by cduartec** [reply]
Hey @samyogshr. I am also getting the same error. Could you share how did you solve it? Thanks :)

**Reply by samyogshr** [reply]
<<The problem was with 'pbs' module, type 'module list' in the terminal to see the currently loaded modules, if 'pbs' is one of them, these steps might help.>>

Steps:

1.  In '.bash_profile', add the following 3 lines at the end.
    
module purge
module load use.moose PETSc
conda deactivate

<<I had activated 'conda' while trying to install, if you haven't you can avoid the third line>>

2. Save, logout from HPC and log back in

3. Then, follow these steps: (Note: I was specifically installing MASTODON)  

mkdir ~/projects4
cd ~/projects4
git clone https://github.com/idaholab/mastodon.git
cd mastodon
git checkout master
git submodule update --init moose

cd moose/scripts/
JOBS=8 ./update_and_rebuild_libmesh.sh
cd ~/projects4/mastodon

make -j 8
./run_tests -j 8

4. If most tests pass, problem solved.

Thank you @hoffwm for your help on this, credit to him.
Samyog
